sdk/appcenter-crashes/src/main/java/com/microsoft/appcenter/crashes/Crashes.java

```diff
@@ -837,10 +837,12 @@ public void run() {
                         ErrorLogReport errorLogReport = unprocessedEntry.getValue();
                         if (errorLogReport.report.getThrowable() instanceof NativeException) {
                             Exception exception = errorLogReport.log.getException();
-                            dumpFile = new File(exception.getStackTrace());
-                            exception.setStackTrace(null);
-                            byte[] logfileContents = FileManager.readBytes(dumpFile);
-                            dumpAttachment = ErrorAttachmentLog.attachmentWithBinary(logfileContents, "minidump.dmp", "application/octet-stream");
+                            if (exception.getStackTrace() != null) {
+                                dumpFile = new File(exception.getStackTrace());
+                                exception.setStackTrace(null);
```

Comment:
This is the only place that set it to null.
As in line 583 the API we use is `@NonNull`. 
This suggests some kind of double processing even though we remove the entry in the iterator.

Reply:
If we can't find why, I would rather we remove `exception.setStackTrace(null)` rather than checking for null as that is only here to cleanup the json we send. That way we can check if that is the cause of the null pointer (double processing) and reopen investigation if crash reported again. Just adding the if will hide problems that will be even harder to investigate.